### PR TITLE
Enable support for PCF

### PR DESCRIPTION
**Pull request by phazebroek**
PCF is different from AWS in that apps are deployed to their own application server instances in the root of the context path. Unlike AWS where the apps have the name of the app as their context root.

Also in PCF there is no unified 'capability' URI to access services. So each service needs it's own configuration / property to specify the URI for that service.

In order to support both AWS and PCF I have refactored this configuration and introduced a new spring profile called 'pcf'.  This configuration is now loaded into a configuration properties class by Spring and is injected/autowired to the rest clients which have been refactored to support this new setup of configuration.

Basically for PCF you only need to specify 'pcf.space=your_space_name' or `environment.name=your_autoconfig_env_name` en add -Dspring.profiles.active=pcf if you want to activate the PCF config.

This PR is still work in progress. Some things still left open todo:
- Clean up code after refactoring (bare minimum to make it work is done only)
- Update readme and documentation accordingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,19 +35,31 @@ It is based on REST and relies on DBS service specs.
...
+- For environments based on AWS (default):
 ```
 java -Denvironment.name=your-env-00 -jar bb-fuel-{version}-boot.jar
 ```
-- For environments based on this configuration: [local.properties](src/main/resources/local.properties) which is based on the [Blade](https://start.backbase.com/) environment setup:
+Note: replace `your-env-00` with the name of your environment. You can omit this property if you've 
+
+- For environments based on local:
+```
+java -Dspring.profiles.active=pcf -jar bb-fuel-{version}-boot.jar
```
Comment (richard-blank) [inline README.md]: -Dspring.profiles.active=**local**

File: src/main/resources/application-pcf.yml [hunk 2]
```diff
@@ -1,11 +1,13 @@
 # Unlike autoconfig, in PCF services are deployed to their own application server instance and have no context path (root).
-# Each service has it's own route which can be configured below.
-# As a rule of thumb, routes contain the name of the space and org to avoid collisions.
+# Each capability' integration service has it's own route which can be configured below.
```
Comment (richard-blank) [inline src/main/resources/application-pcf.yml]: Each capability **specific** integration service has **its** own route....

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -13,18 +11,31 @@ logging:
...
+eureka.client.enabled: false
+
+environment:
+  # Change name of env:
+  name: your-env-00
+  domain: backbase.test:8080
+
+bb-fuel:
+  platform:
+    infra: ${environment.name}-infra
```
Comment (richard-blank) [inline src/main/resources/application.yml]: infra suffix could be editorial: 
e.g. ${environment.name}-${infra.base:infra}

File: README.md [hunk 4]
```diff
@@ -85,6 +102,7 @@ bb-fuel supports multiple DBS versions. See below which version maps to the requ
 
 | DBS version | bb-fuel minimal [version](https://github.com/backbase/bb-fuel/releases) |
 |-------------|-------------------------------------------------------------------------|
+| 2.14.1 PCF  | 2.0.0+                                                                  |
 | 2.14.1      | 1.8.20+                                                                 |
```
Comment (richard-blank) [inline README.md]: PCF is not a version, i would make it: 2.14.1      | 1.8.20,2.0.0+

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -20,7 +20,8 @@ security.basic.enabled: false
 eureka.client.enabled: false
 
 environment:
-  name: lingering-sonic-21
+  # Change name of env:
```
[the remark this answers is not in the document]
Comment (phazebroek) [reply src/main/resources/application.yml]: There is a git command to ignore local changes, but include new changes from remote. I don't know the command by heart but maybe we can include it in a comment in this file or so, so changes are not included/accidentally committed.

[the remark this answers is not in the document]
Comment (phazebroek) [reply src/main/resources/application.yml]: Removed btw.
